presto-main/src/main/java/com/facebook/presto/sql/planner/RelationPlanner.java

```diff
@@ -299,6 +301,23 @@ else if (firstDependencies.stream().allMatch(right.canResolvePredicate()) && sec
                 Optional.empty(),
                 Optional.empty());
 
+        if (node.getType() != INNER) {
+            List<InPredicate> inPredicateSubqueries = analysis.getInPredicateSubqueries(node);
+            for (Expression complexExpression : complexJoinExpressions) {
+                for (InPredicate inPredicate :  inPredicateSubqueries) {
+                   if (nodeContains(complexExpression, inPredicate)) {
+                       throw new SemanticException(
+                               NOT_SUPPORTED,
+                               inPredicate,
+                               "IN with subquery predicate in join condition not supported");
+                   }
+                }
+            }
+
+            // non inner join subqueries can be applied only to one side of join
+            leftPlanBuilder = subqueryPlanner.handleSubqueries(leftPlanBuilder, complexJoinExpressions, node);
```

Comment:
I will rewrite the comment as it does not tell much.

There is not much special about the left side. It could be a right, not much difference. It is for any kind of outer join. 

Subquery (non LATERAL JOIN) cannot access FROM relations. `complexExpression` have to have access to all FROM relations as well as to subquery, so subquery has to be added to the one side of join. Left is selected in arbitrary way.